optimum/habana/transformers/generation/utils.py

```diff
@@ -382,16 +511,28 @@ def generate(
                 inputs_tensor, generation_config.pad_token_id, generation_config.eos_token_id
             )
 
-        if generation_config.static_shapes:
+        if self.config.is_encoder_decoder and generation_config.static_shapes:
```

Comment:
Inputs and attention mask for decoder-only models will not be padded if you do this, is it intended?

Reply:
No. I've made the change to avoid inputs and attention_mask padding in case of encoder_decoder models as inputs are already padded. Where as in decoder models, Inputs are not padded